packages/docs/concepts/menus.yaml

```diff
@@ -0,0 +1,156 @@
+                    links:
+                      - id: page2
+                        type: MenuLink
+                        pageId: page2 # pageId will be used as link title
+                      - id: external
+                        type: MenuLink
+                        url: https://external.com
+                        properties:
+                          title: External site
+                      - id: nested-group
+                        type: MenuGroup
+                        properties:
+                          title: Nested Group
+                        links:
+                          - id: page3
+                            type: MenuLink
+                            pageId: page3
+                            properties:
+                              title: Page 3
+              - id: page-1-and-3
+                links:
+                  - id: page1
+                    type: MenuLink
+                    pageId: page1
+                    properties:
+                      title: Page 1
+                      icon: AiOutlineFile
+                  - id: page3
+                    type: MenuLink
+                    pageId: page3
+                    properties:
+                      title: Page 3
+                      icon: AiOutlineControl
+            ```
+
+            We can make use of [protected pages](/protected-pages) and [roles](/roles) to filter the menu items according to the pages that the user has access to.
+
+            ###### lowdefy.yaml
+            ```yaml
+            lowdefy: LOWDEFY_VERSION
+
+            auth:
+              # ...
+              pages:
+                protected: true
+                public:
+                  - login
+                  - '404'
+                roles:
+                  admin:
+                    - page2
+                    - page3
+                  user:
+                    - page1
+
+            menus:
+              - id: default
+                links:
+                  - id: page1
+                    type: MenuLink
+                    pageId: page1
+                    properties:
+                      title: Page 1
+                      icon: AiOutlineFile
+                  - id: page2
+                    type: MenuLink
+                    pageId: page2 # pageId will be used as link title
+                  - id: page3
+                    type: MenuLink
+                    pageId: page3
+                    properties:
+                      title: Page 3
+              ```
+
+              Public users can only view the `login` and `404` pages, all other pages require the user to be logged in. If the authorized user had been given only the `user` role, they will only be able to see `page1`. In order to see `page2` and `page3`, the authorized user will need to have the `admin` role assigned to them. The menu will be filtered accordingly and will only contain the menu items that the user has access to.
```

Comment:
In the example above, public users can only ...

Reply:
Change made in 0f8e3ebcba1315a5f14cec4a02cd6f8426f948e2